{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#グラフ理論\n",
    "\n",
    "1.Introduction\n",
    "\n",
    "Biological domains where graph theory techniques are applied for knouledge extraction from data.\n",
    "\n",
    "1. Modeling of bio-molecular networks.→3\n",
    "   -protein interaction networks, metabolic networks, tanscriptional regulatory networks\n",
    "  \n",
    "2. Measurement of centrality and importance in bio-molecular networks.→4\n",
    "\n",
    "3. Identifying motifs or functional modules in biological networks.→5\n",
    "\n",
    "4. Mining novel pathways from bio-molecular networks.→6\n",
    "\n",
    "\n",
    "2.Definitions and mathematical preliminaries\n",
    "\n",
    "2.1 The concept of graph\n",
    "Graph G : a set of vertices V(G) and a set of edges E(G)\n",
    "The number of vertices: |V(G)|\n",
    "The set of vertices adjacent to a vertex v1: N(v1)\n",
    "The degree of a vertex v1: the number of edges with which it is incident d(v1)\n",
    "Subgraph G': all vertices and edges are included in G. G is supergraph of G'.\n",
    "\n",
    "2.2 Directed and undirected graphs\n",
    "Directed graph: no distinction between the two vertices associated with each edge.\n",
    "Undirected graph: its edges are directed from one vertices to aother.\n",
    "\n",
    "2.3 Node-degree and the adjacency matrix\n",
    "・Undirected graph\n",
    "d(u): the degree of node u in V(G) = the total number of edges at u = |N(u)|\n",
    "・Directed graph\n",
    "d+(u): the number of edges that terminate at u.\n",
    "The adjacensy matrix A: 点iと点jを結ぶ辺の本数を第ij要素とするn × nの行列\n",
    "\n",
    "2.4 Path, path length and connecter graph\n",
    "path length: distance d(u,v)\n",
    "connected graph: there is some path from u to v, for every pair of vertices,(u,v)\n",
    "\n",
    "3. Modeling of bio-molecular networks\n",
    "・transcriptional regulatory networks：directed graphs\n",
    "・protein interaction network：undirected graphs\n",
    "・metabolic network：directed graphs\n",
    "\n",
    "4. Measurement of centrality and importance in bio-molecular networks\n",
    "Modules: independent from other modules\n",
    "How modules and their interconnections are changed by evolution; selection process?\n",
    "Interacting pairs of genes: lie in alternate pathways rather than cluster, do not interact directly\n",
    "A detailed functional analysis of the network by simulating knockdown or over-expression experiments is possible.\n",
    "Silico evolution: metabolic, signaling, biochemical, regulatory, electronic, neural networks\n",
    "\n",
    "5. Identifying motifs or functional modules in biological networks\n",
    "Biological systems - engineering systems: flow charts\n",
    "share structural principals, both are organized with modularity\n",
    "・Engineering systems\n",
    "　 decomposed into functional modules at different levels, subroutines in software, replaceble parts in machines\n",
    "・Biological networks\n",
    " 　no consensus on the precise groups of genes and interactions that form modules\n",
    " 　possess a modular structure\n",
    "  \n",
    "A module in a network: a set of nodes that have strong interactions and a common function\n",
    " input nodes \n",
    " output nodes\n",
    "\n",
    "Basic functional modules\n",
    "Motifs: basic building blocks that may have specific functions as elementary computational circuits\n",
    "        closely related to conventional building blocks\n",
    "        \n",
    "No adequate and precise analysis\n",
    "It is not clear what determines the particular frequencies of all possible network motifs\n",
    "\n",
    "6. Mining novel pathways from bio-molecular networks\n",
    "A minimal set of pathways required by all organisms\n",
    "What extent are the genomic pathways conserved among different species\n",
    "How are organisms related in terms of the distance between pathways\n",
    "\n",
    "The identification of pathways in different orgasms\n",
    "Predict pathways in an organism's biological networks\n",
    "sabstrates - products\n",
    "Redefining microbial biochemical pathways based on missing protein\n",
    "\n",
    "7. Conclusion\n",
    "Network graphs have the advantage that they are very simple to reason about, and correspond by and large to the information that is globally available today on the network level.\n",
    "A comprehensive understanding of these networks is needed to develop more sophisticated and effective treatment strategies for diseases.\n",
    "Mathematical models of largescale data sets are valuable in medical problems.\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
